Sources/Apollo/NormalizedCache.swift

```diff
@@ -1,5 +1,50 @@
 import Foundation
 
+/// An object that drives behavior of clear operations on a given cache.
+public struct CacheClearingPolicy: Equatable {
+  /// Clears all records in the cache.
+  public static let allRecords = CacheClearingPolicy(.allRecords)
+  /// Clears all records whose key matches the provided glob pattern.
+  /// - Parameter pattern: The glob pattern to use for key matching. For example `*pollo` will match both `Apollo` and `pollo`.
+  public static func allMatchingKeyPattern(_ pattern: String) -> CacheClearingPolicy { .init(.allMatchingKeyPattern(pattern)) }
+  /// Clears the the first (oldest) records in the cache up to the given limit.
+  /// - Parameter k: The number of records to remove from the cache.
+  public static func first(_ k: Int) -> CacheClearingPolicy { .init(.first(k)) }
```

Comment:
I can't think of a use case where `last` and `first` are useful.

Reply:
This was brought in from the implementation found in the public branch that is opened against Apollo proper: https://github.com/apollographql/apollo-ios/pull/1512

I was just trying to be thorough as early as possible with as many possible ways to clear the cache